### PR TITLE
Use the DOTNET_ env var prefix

### PR DESCRIPTION
**Pull request by Tratcher**
#4150
 @glennc

### CODE REVIEW

**Comment by davidfowl** [comment]
Needs a test

File: src/Hosting/Hosting/src/Host.cs [hunk 1]
```diff
@@ -20,7 +20,7 @@ public static IHostBuilder CreateDefaultBuilder(string[] args)
             builder.UseContentRoot(Directory.GetCurrentDirectory());
             builder.ConfigureHostConfiguration(config =>
             {
-                config.AddEnvironmentVariables(); // TODO: Prefix?
+                config.AddEnvironmentVariables("DOTNET_");
```
Comment (JunTaoLuo) [inline src/Hosting/Hosting/src/Host.cs]: nit: use a named variable for the prefix like we do everywhere else: https://github.com/aspnet/AspNetCore/blob/c458fe6ebe24ab94f098a453825e71207721b9b9/src/Hosting/Hosting/src/WebHostBuilder.cs#L44

**Comment by Tratcher** [comment]
@davidfowl reliably testing env vars needs out-of-proc functional testing. We don't have that infrastructure in Extensions and it isn't worth building for this.

**Comment by davidfowl** [comment]
@Tratcher we need more than 0 coverage. I think it would be reasonable to write those tests in ASP.NET Core hosting since we do that the infrastructure on that side.

**Comment by Tratcher** [comment]
Or part of the template I'm working on.

**Pull request merged**